client/src/glossary/GlossaryMenu.tsx

```diff
@@ -0,0 +1,149 @@
+
+  constructor(props: GlossaryMenuProps) {
+    super(props);
+
+    this.state = {
+      show: this.props.show,
+      results: [],
+    };
+  }
+
+  static getDerivedStateFromProps(
+    nextProps: GlossaryMenuProps,
+    prevState: GlossaryMenuState
+  ) {
+    const { show: next_show } = nextProps;
+    const { show: prev_show } = prevState;
+
+    if (next_show !== prev_show) {
+      return {
+        show: next_show,
+      };
+    } else {
+      return null;
+    }
+  }
+
+  getResults = (childData: ResultProps[]) => {
+    this.setState({
+      results: childData,
+    });
+  };
+
+  closeItem() {
+    this.props.setGlossaryItem({});
+  }
+
+  openItem(item: Record<string, unknown>) {
+    this.props.setGlossaryItem(item);
+  }
+
+  handleKeyPress(e: React.KeyboardEvent<HTMLSpanElement>) {
+    if (e.key === "Enter") {
+      this.props.toggle(false);
+    }
+  }
+
+  render() {
+    return (
+      <div
+        className={
+          this.state.show
+            ? "glossary-sidebar-wrapper active"
+            : "glossary-sidebar-wrapper"
+        }
+      >
+        <aside
+          role="dialog"
+          aria-labelledby="glossary-title"
+          className="glossary-sidebar"
+          ref={this.main}
+        >
+          <div className="glossary-sidebar-header-wrapper" ref={this.header}>
+            <div className="glossary-sidebar-header">
+              <div role="navigation" aria-label="Glossary navigation">
+                <div className={"close-button"}>
+                  <span
+                    role="button"
+                    className="icon"
+                    onClick={() => this.props.toggle(false)}
+                    onKeyDown={(e) => this.handleKeyPress(e)}
+                    tabIndex={0}
+                  >
+                    <IconX width="25px" color="white" alternate_color={false} />
+                  </span>
+                </div>
+              </div>
+              <h1 id="glossary-title" className="glossary-title" tabIndex={-1}>
+                Glossary
+              </h1>
+              <div className="search-wrapper">
+                <SearchConfigSidebar
+                  placeholder={"testing"}
+                  search_configs={[glossary_search_config]}
+                  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                  getResults={(data: ResultProps[]) => this.getResults(data)}
+                />
+              </div>
+              <div className="glossary-example">
+                Example: &quot;Capital Vote&quot;
+              </div>
+            </div>
+          </div>
+          <div className="glossary-sidebar-content-wrapper">
+            <SidebarContent
+              title={this.props.item.title}
+              def={this.props.item.def}
+              results={this.state.results}
+              closeItem={() => this.closeItem()}
+              openItem={(item: Record<string, unknown>) => this.openItem(item)}
```

Comment:
RE suggestion to just pass around glossary keys around rather than `item`s: here's an example of where that'd make things easier. Simple string type, instead of needing (I assume just for dev expediency) a mystery record.